orttraining/orttraining/training_ops/cuda/communication/recv.cc

```diff
@@ -40,113 +135,108 @@ Status Recv::ComputeInternal(OpKernelContext* ctx) const {
 
 #ifdef ENABLE_NVTX_PROFILE
   profile::NvtxRangeCreator preRange(
-    "PreRecv-" + std::to_string(src), profile::Color::Green);
+      "PreRecv-" + std::to_string(src), profile::Color::Green);
   // Begin of preparation for receiving data.
   preRange.Begin();
 #endif
 
-  // Create buffers
-  const int tensor_num = static_cast<int>(element_types_.size());
-  // TODO move the following variables to member variables for extending life-time
-  // if we want to make the entire call async
-  std::vector<size_t> prefix_tensor_shape_sizes(tensor_num);
-  std::vector<int64_t> aggregated_tensor_shapes;
-  size_t aggregated_aligned_tensor_bytes = 0;
-
   // Start communication
   int world_rank;
-  MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
+  MPI_CHECK(MPI_Comm_rank(MPI_COMM_WORLD, &world_rank));
   ORT_ENFORCE(world_rank != src, "Receive data from rank ", src, " on the rank ", world_rank, ".");
```

Comment:
>ORT_ENFORCE(world_rank != src, "Receive data from rank ", src, " on the rank ", world_rank, ".") [](start = 2, length = 96)

Maybe just return with error status when communication is not set up correctly

Reply:
Will replace it with MPI_CHECK which is the default check for MPI in ORT.

---
In reply to: [462634292](https://github.com/microsoft/onnxruntime/pull/4547#discussion_r462634292) [](ancestors = 462634292)